File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
       int res = 0;
        for(int i=0;i<s.size();i++)
        {
            vector<bool>visited(256);
            for(int j=i;j<s.size();j++)
            {
                if(visited[s[j]] == true)
                {
                    break;
                }
                else
                {
                    res = max(res,j-i+1);
                    visited[s[j]] = true; 
                }
            }
        }
        return res;
        
        
    }
};

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start=0
        longest = 0
        dick = {}
        for i in range(len(s)) :
            if s[i] in dick and start <= dick[s[i]]:
                start = dick[s[i]] + 1
            else :
                longest = max(longest,i - start + 1)
            dick[s[i]] = i 
        return longest
                
                
                                        
        